### DISCUSSION TITLE
Recommendations for interface with radio automation

### DISCUSSION THREAD

**Discussion by whmountains** [savonet/liquidsoap#2736] [Ideas]
I want to create a new radio automation software and use liquidsoap as the playback engine.  In doing so I am following in the footsteps of software like Libretime and Azuracast.  In fact, my first introduction to liquidsoap was back when Airtime was open source and I was trying to install it on an OG Raspberry Pi B.

My question is, what are your recommendations for the interface between the automation software and liquidsoap?  I can't just restart liquidsoap with a new script, because that would interrupt playback.  So the telnet server would need to be involved somehow.  Maybe I am underestimating the power of the telnet server, but I think I want something higher-level.  How can I dynamically instruct liquidsoap to switch to playlist X at Y time for example?

I would like to see more activity in the open-source radio automation space.  Radio automation basically has three parts: scheduling, media management, and GUI.  In my opinion, scheduling is the least accessible part.  Liquidsoap can do all of this, and even has a flexible telnet server, but to a web developer that is strange and difficult.  We are most comfortable with REST APIs and pub-sub channels.  I would like to figure out a way to make liquidsoap's powerful scheduling primitives available over something like a REST API or a websocket.

I'm not sure how much of this should go inside liquidsoap and how much should be in a separate program, but the end goal would be to create a "scheduling engine" that any web developer would feel comfortable interfacing with while bringing their own take on media management and UI design.

**Comment by vitoyucepi** [comment]
Hi @whmountains,
I've got another sample of radio automation for you. https://gitlab.servus.at/aura

You can develop your API and interact with liquidsoap using HTTP+JSON.
Though you could serve JSON over the telnet if you want.

I think that the WebSocket interface is pretty limited. https://github.com/webcast/webcast.js

**Comment by add-n2x** [comment]
@vitoyucepi Thanks for sharing the link to AURA (Automated Radio). 

@whmountains We also have a minimal website with links to the docs and repos at https://aura.radio. There is also an [simplified overview](https://docs.aura.radio/en/latest/administration/overview.html).

In fact, AURA is following in the footsteps of Airtime and LibreTime. We are a bunch of Austrian and German community radios, which are in the need of an Open Source Radio Automation Suite. Many of us are actually using software like Airtime, but due to the nature of community radios, missing some "community aspects" in those tools. That's why we are building Automated Radio.

We are currently in alpha, so don't expect a production ready functionality yet. But we are happy hearing about your needs, while keep building. If you are developer, we are also happy about contributions. So let's keep in touch. :)

> Radio automation basically has three parts: scheduling, media management, and GUI.

The calendar information is stored in _Steering_, a Django application holding all radio data, like show, hosts, episode, schedule and timeslot information. The _Dashboard_ is the VueJS frontend to manage the data. _Tank_ is in charge of uploading and normalizing audio files. And managing playlists. And there is _Engine_, for scheduling, playout and recording.

> In my opinion, scheduling is the least accessible part. Liquidsoap can do all of this, and even has a flexible telnet server, but to a web developer that is strange and difficult. We are most comfortable with REST APIs and pub-sub channels. I would like to figure out a way to make liquidsoap's powerful scheduling primitives available over something like a REST API or a websocket.

Anyway, if your are not interested in a fully fledged radio software suite, and just want to get some ideas on how you could automate Liquidsoap yourself, checkout _Engine_. This is a Python scheduling and control engine to interact with Liquidsoap via file sockets (_Engine Core_). It's the same interface as with the Telnet server. There is also a REST API available to query playlogs.
